src/main/java/org/mule/extension/file/common/api/BaseFileSystemOperations.java

```diff
@@ -298,6 +331,39 @@ protected void doWrite(FileConnectorConfig config, FileSystem fileSystem, String
     fileSystem.write(path, content, mode, lock, createParentDirectories);
   }
 
+  /**
+   * Writes the {@code content} into the file pointed by {@code path}.
+   * <p>
+   * If the directory on which the file is attempting to be written doesn't exist, then the operation will either throw
+   * {@link IllegalArgumentException} or create such folder depending on the value of the {@code createParentDirectory}.
+   * <p>
+   * If the file itself already exists, then the behavior depends on the supplied {@code mode}.
+   * <p>
+   * This operation also supports locking support depending on the value of the {@code lock} argument, but following the same
+   * rules and considerations as described in the read operation.
+   *
+   * @param config                  the {@link FileConnectorConfig} on which the operation is being executed
+   * @param fileSystem              a reference to the host {@link FileSystem}
+   * @param path                    the path of the file to be written
+   * @param content                 the content to be written into the file. Defaults to the current {@link Message} payload
+   * @param createParentDirectories whether or not to attempt creating any parent directories which don't exists.
+   * @param lock                    whether or not to lock the file. Defaults to false
+   * @param mode                    a {@link FileWriteMode}. Defaults to {@code OVERWRITE}
+   * @param lockTimeout             time in milliseconds that the operation will spend trying to lock the file.
+   * @throws IllegalArgumentException if an illegal combination of arguments is supplied
+   */
+  protected void doWrite(FileConnectorConfig config, FileSystem fileSystem, String path, InputStream content,
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Method has 8 parameters, which is greater than 7 authorized. ([squid:S00107](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS00107))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule-file-commons&pullRequest=56&issues=AWm2SxsLotIXnmsUPRcD&open=AWm2SxsLotIXnmsUPRcD)

[ISSUE_KEY]: AWm2SxsLotIXnmsUPRcD